compiler/rustc_hir_typeck/src/upvar.rs

```diff
@@ -361,7 +361,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         // For coroutine-closures, we additionally must compute the
         // `coroutine_captures_by_ref_ty` type, which is used to generate the by-ref
         // version of the coroutine-closure's output coroutine.
-        if let UpvarArgs::CoroutineClosure(args) = args {
+        if let UpvarArgs::CoroutineClosure(args) = args
+            // Don't do this if we are tainted by errors, because fallback causes us
+            // to fail to infer upvars for the outer coroutine-closure.
+            && self.tainted_by_errors().is_none()
```

Comment:
Is there a more local thing you can check? `args.errors_reported()` or some other list that isn't item-global?

Reply:
Yeah, we can check for ReError in the args instead (we just need to resolve vars) -- I was just using the same criteria as used in fallback.

Reply:
Ah... yea that sadly makes sense